x/src/test.rs

```diff
@@ -0,0 +1,142 @@
+use crate::config::Package;
+use crate::{
+    config::Config,
+    utils::{self, project_root},
+    Result,
+};
+use anyhow::anyhow;
+use std::iter;
+use std::process::{Command, Stdio};
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+pub struct Args {
+    #[structopt(long, short, number_of_values = 1)]
+    /// Run test on the provided packages
+    package: Vec<String>,
+    #[structopt(long, short)]
+    /// Only run unit tests
+    unit: bool,
+}
+
+pub fn run(args: Args, config: Config) -> Result<()> {
+    if args.unit {
+        run_cargo_test_on_packages_separate(
+            config
+                .package_exceptions()
+                .iter()
+                .filter(|(_, pkg)| !pkg.system)
+                .map(|(p, pkg)| (p.as_str(), pkg)),
+        )?;
+        run_cargo_test_on_most_things(config.package_exceptions().iter().map(|(p, _)| p.as_str()))?;
```

Comment:
I think we should rename this `run_cargo_test_with_exclusions`. I had to read the implementation to understand why the first argument was passing the exceptions.

Reply:
Yep that's a good idea.